pkg/assembler/backends/arangodb/certifyGood.go

```diff
@@ -889,50 +889,11 @@ func (c *arangoClient) IngestCertifyGoods(ctx context.Context, subjects model.Pa
 	}
 }
 
-func getPkgCertifyGood(ctx context.Context, cursor driver.Cursor) ([]*model.CertifyGood, error) {
-	type collectedData struct {
-		PkgVersion    *dbPkgVersion `json:"pkgVersion"`
-		CertifyGoodID string        `json:"certifyGood_id"`
-		Justification string        `json:"justification"`
-		Collector     string        `json:"collector"`
-		Origin        string        `json:"origin"`
-	}
-
-	var createdValues []collectedData
-	for {
-		var doc collectedData
-		_, err := cursor.ReadDocument(ctx, &doc)
-		if err != nil {
-			if driver.IsNoMoreDocuments(err) {
-				break
-			} else {
-				return nil, fmt.Errorf("failed to package certifyGood from cursor: %w", err)
-			}
-		} else {
-			createdValues = append(createdValues, doc)
-		}
-	}
-
-	var certifyGoodList []*model.CertifyGood
-	for _, createdValue := range createdValues {
-		pkg := generateModelPackage(createdValue.PkgVersion.TypeID, createdValue.PkgVersion.PkgType, createdValue.PkgVersion.NamespaceID, createdValue.PkgVersion.Namespace, createdValue.PkgVersion.NameID,
-			createdValue.PkgVersion.Name, createdValue.PkgVersion.VersionID, createdValue.PkgVersion.Version, createdValue.PkgVersion.Subpath, createdValue.PkgVersion.QualifierList)
-
-		certifyGood := &model.CertifyGood{
-			ID:            createdValue.CertifyGoodID,
-			Subject:       pkg,
-			Justification: createdValue.Justification,
-			Origin:        createdValue.Collector,
-			Collector:     createdValue.Origin,
-		}
-		certifyGoodList = append(certifyGoodList, certifyGood)
-	}
-	return certifyGoodList, nil
-}
-
-func getArtifactCertifyGood(ctx context.Context, cursor driver.Cursor) ([]*model.CertifyGood, error) {
+func getCertifyGoodFromCursor(ctx context.Context, cursor driver.Cursor) ([]*model.CertifyGood, error) {
 	type collectedData struct {
+		PkgVersion    *dbPkgVersion   `json:"pkgVersion"`
 		Artifact      *model.Artifact `json:"artifact"`
+		SrcName       *dbSrcName      `json:"srcName"`
```

Comment:
nit: wonder if its possible to do an embedded struct here and so to dedup some of the matching code.

Reply:
hmmm, actually can you expand upon this more? I don't understand where to make the changes.

Reply:
yea so im thinking that the matching against pkg, src or artifact is kind of a repeated pattern in the collectedData struct, so wondering we can embed it such as having
```
Artifact
SrcName
PackageName
PackageVersion
```
as an embedded struct and just including it in each collectedData struct, and then having the parsing function https://github.com/guacsec/guac/pull/1184/files#diff-d396bee1ffad665a5bbc2762f9391519ffda2dd538a5b222131d6b68b1731794R920-R929 be refactored to a function so we don't have to repeat it multiple times.


This would be something probably not within this PR but could be a nice QOL overall since there's quite some room for reuse.